R/rma-tidiers.R

```diff
@@ -0,0 +1,222 @@
+#' Tidying methods for meta-analyis objects
+#'
+#' These methods tidy the results of meta-analysis objects from the metafor package
+#'
+#' @param x An `rma` created by the `metafor` package.
+#' @inheritParams tidy.lm
+#' @param include_studies Logical. Should individual studies be included in the
+#'    output?
+#' @param ... Additional arguments
+#' @param measure Measure type. See [metafor::rma()]
+#'
+#' @return A `tibble`
+#' @export
+#'
+#' @examples
+#' 
+#' library(metafor)
+#' 
+#' df <-
+#'   escalc(
+#'     measure = "RR",
+#'     ai = tpos,
+#'     bi = tneg,
+#'     ci = cpos,
+#'     di = cneg,
+#'     data = dat.bcg
+#'   )
+#' 
+#' meta_analysis <- rma(yi, vi, data = df, method = "EB")
+#'
+#' tidy(meta_analysis)
+#'
+#' @rdname tidiers
+#' 
+tidy.rma <- function(x, conf.int = FALSE, conf.level = 0.95, exponentiate = FALSE,
+                     include_studies = TRUE, measure = "GEN", ...) {
+  
+  estimates <- metafor::escalc(yi = x$yi.f, vi = x$vi.f, measure = measure) %>%
+    summary(level = conf.level * 100) %>% 
+    as.data.frame(stringsAsFactors = FALSE)
+  
+  estimates <- cbind(x$slab, "study", estimates[, c("yi", "sei", "zi")], NA,
```

Comment:
Can I ask why the `NA` here? Also, would you consider using `bind_cols()` instead?

Reply:
`NA` is for the p values for the individual studies, which aren't in the model. Only the summary estimate returns a p value. I don't mind switching to `bind_cols()`, but I'll have to change a few things since it's strict about all the objects having the same length